packages/core/src/contrib/Constraints.ts

```diff
@@ -971,41 +936,42 @@ export const overlap1D = (
 };
 
 /**
- * Return the bounding box (as 4 points) of an axis-aligned box-like shape given by `center`, width `w`, height `h` as an object with `minX, maxX, minY, maxY`.
+ * Input: A rect-like shape.
+ * Output: A bbox corresponding to the rect.
  */
-export const bbox = (center: VecAD, w: VarAD, h: VarAD): any => {
-  const halfWidth = div(w, constOf(2.0));
-  const halfHeight = div(h, constOf(2.0));
-  const nhalfWidth = neg(halfWidth);
-  const nhalfHeight = neg(halfHeight);
-  // CCW: TR, TL, BL, BR
-  const pts = [
-    [halfWidth, halfHeight],
-    [nhalfWidth, halfHeight],
-    [nhalfWidth, nhalfHeight],
-    [halfWidth, nhalfHeight],
-  ].map((p) => ops.vadd(center, p));
+export const bbox = ([t, s]: [string, any]): BBox => {
+  if (!(isRectlike(t) || isLinelike(t))) {
+    throw new Error(
+      `Bbox expected a rect-like or line-like shape, but got ${t}`
+    );
+  }
 
-  const rect = {
-    minX: pts[1][0],
-    maxX: pts[0][0],
-    minY: pts[2][1],
-    maxY: pts[0][1],
-  };
+  // TODO: Compute the bbox of the line in a nicer way
 
-  return rect;
-};
+  // initialize center, w, and h depending on whether the input shape is line-like or rect/square-like
+  const center = isLinelike(t)
```

Comment:
ternaries here. is it worth weakening/possible to weaken `const` to `let` and replace with if-else to improve readability?

Reply:
Doesn't matter here, as using native JS ternaries is not an autodiff operation, and won't show up in the graph or during the optimization in runtime. (This is fine because a shape's type doesn't change during optimization.) I would go for the more readable option.